python-avd/pyavd/_eos_designs/shared_utils/utils.py

```diff
@@ -166,3 +167,17 @@ def match_nodes(self: SharedUtilsProtocol, nodes: Sequence[str]) -> bool:
         if not nodes:
             return True
         return self.match_regexes(nodes, self.hostname)
+
+    @cached_property
+    def underlay_vlan_trunk_groups(self: SharedUtilsProtocol) -> dict[int, set[str]]:
+        """Return an EosCliConfigGen.Vlans object containing all the underlay VLAN with their trunk groups."""
+        vlans: dict[int, set[str]] = {}
+        for peer in self.switch_facts.downlink_switches:
+            peer_facts = self.get_peer_facts(peer)
+            for uplink in peer_facts.uplinks:
```

Comment:
We could just take the first uplink (if any) and break afterwards, since we know that all uplinks are created equal. Actually we could move the peer_trunk_groups out of the individual uplinks and to a common `uplink_switch_trunk_groups` or something.